### PR TITLE
Simplify display of verification requests in timeline

### PR DESCRIPTION
**Pull request by andybalaam**
Fixes https://github.com/vector-im/element-web/issues/26450 . This change only affects the `m.key.verification.request` type - there will be another for `m.key.verification.concusion`.

This change affects both Element R and legacy crypto. It reduces the functionality in legacy crypto (which arguably unwisely showed changeable status info in a timeline event), and increases the functionality in Element R (which previously showed nothing).

As agreed in https://github.com/vector-im/element-web/issues/26258#issuecomment-1818828928 :

> UX should only show a tile for m.key.verification.request event type, without any info about the outcome (user can just check the profile to get verification state), just who requested, and ignore other events related to that verification.

After this change, we don't show any indication of the outcome of a request, or any buttons to respond:

image
image

Before:
image
image
image
image

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Simplify display of verification requests in timeline ([\#11931](https://github.com/matrix-org/matrix-react-sdk/pull/11931)). Fixes vector-im/element-web#26450. Contributed by @andybalaam.

### CODE REVIEW

**Comment by t3chguy** [comment]
Has this gone through Product or Design? Losing the main place I'd expect to interact with an ongoing request from seems quite big

**Review by pmaier1 — APPROVED** [review]
:+1:

File: src/components/views/messages/MKeyVerificationRequest.tsx [hunk 1]
```diff
@@ -15,187 +15,71 @@ limitations under the License.
...
+    from_device: string;
+    methods: Array<string>;
+    msgtype: "m.key.verification.request";
+    to: string;
+}
 
-    private onRequestChanged = (): void => {
-        this.forceUpdate();
-    };
+const MKeyVerificationRequest: React.FC<Props> = ({ mxEvent, timestamp }) => {
```
Comment (florianduros) [inline src/components/views/messages/MKeyVerificationRequest.tsx:40]: I'm not against a bit of documentation about the component. What it is it's purpose and it should behave :)

**Review by florianduros — APPROVED** [review]
Looks good :)

File: src/components/views/messages/MKeyVerificationRequest.tsx (hunk 1 above)
Comment (andybalaam) [reply src/components/views/messages/MKeyVerificationRequest.tsx:40]: Thanks, good call.

**Pull request merged**